packages/front-end/enterprise/components/Dashboards/DashboardsTab.tsx

```diff
@@ -0,0 +1,567 @@
+import Button from "@/components/Radix/Button";
+import { useAuth } from "@/services/auth";
+import DeleteButton from "@/components/DeleteButton/DeleteButton";
+import usePermissionsUtil from "@/hooks/usePermissionsUtils";
+import MoreMenu from "@/components/Dropdown/MoreMenu";
+import EditButton from "@/components/EditButton/EditButton";
+import { Select, SelectItem, SelectSeparator } from "@/components/Radix/Select";
+import Tooltip from "@/components/Tooltip/Tooltip";
+import { useUser } from "@/services/UserContext";
+import { useCopyToClipboard } from "@/hooks/useCopyToClipboard";
+import { DropdownMenuSeparator } from "@/components/Radix/DropdownMenu";
+import { useDashboards } from "@/hooks/useDashboards";
+import PaidFeatureBadge from "@/components/GetStarted/PaidFeatureBadge";
+import UpgradeModal from "@/components/Settings/UpgradeModal";
+import DashboardEditor from "./DashboardEditor";
+import DashboardSnapshotProvider from "./DashboardSnapshotProvider";
+import DashboardModal from "./DashboardModal";
+
+export type CreateDashboardArgs = {
+  method: "POST";
+  dashboardId?: never;
+  data: {
+    title: string;
+    editLevel: DashboardInterface["editLevel"];
+    enableAutoUpdates: boolean;
+    blocks?: DashboardBlockData<DashboardBlockInterface>[];
+  };
+};
+export type UpdateDashboardArgs = {
+  method: "PUT";
+  dashboardId: string;
+  data: Partial<{
+    title: string;
+    blocks: DashboardBlockInterfaceOrData<DashboardBlockInterface>[];
+    editLevel: DashboardInterface["editLevel"];
+    enableAutoUpdates: boolean;
+  }>;
+};
+export type SubmitDashboard<
+  T extends CreateDashboardArgs | UpdateDashboardArgs
+> = (args: T) => Promise<void>;
+
+export const autoUpdateDisabledMessage =
+  "Automatic updates are disabled for dashboards with Dimension Analyses or SQL Explorer blocks, or when experiment updates are disabled";
+
+interface Props {
+  experiment: ExperimentInterfaceStringDates;
+  initialDashboardId: string;
+}
+
+export default function DashboardsTab({
+  experiment,
+  initialDashboardId,
+}: Props) {
+  const [dashboardId, setDashboardId] = useState(initialDashboardId);
+  const [dashboardCopy, setDashboardCopy] = useState<
+    DashboardInterface | undefined
+  >(undefined);
+  const [hasMadeChanges, setHasMadeChanges] = useState(false);
+  useEffect(() => {
+    if (initialDashboardId) {
+      setDashboardId(initialDashboardId);
+    }
+  }, [initialDashboardId]);
+  const { dashboards, mutateDashboards } = useDashboards(experiment.id);
+  const defaultDashboard = dashboards.find((dash) => dash.isDefault);
+
+  useEffect(() => {
+    if (!dashboardId && dashboards.length > 0) {
+      setDashboardId(defaultDashboard?.id ?? dashboards[0].id);
+    }
+  }, [dashboardId, dashboards, defaultDashboard]);
+
+  const {
+    userId,
+    settings: { updateSchedule },
+  } = useUser();
+  const [isEditing, setIsEditing] = useState(false);
+  const [showUpgradeModal, setShowUpgradeModal] = useState(false);
```

Comment:
question: When we show the upgrade modal how can we ensure the user understands this is only in the Enterprise plan and not included in Pro?